src/libostree/ostree-repo.c

```diff
@@ -4905,6 +4906,44 @@ ostree_repo_load_commit (OstreeRepo            *self,
                                  out_variant, NULL, NULL, out_state, NULL, error);
 }
 
+GHashTable *
+ostree_repo_list_objects_set (OstreeRepo                  *self,
```

Comment:
Can we dedupe this with `ostree_repo_list_objects`? E.g. have the latter call this one with `dummy_value` passed in and in the other callsite, pass `NULL`.

Reply:
There wasn't *that* much code in both, but OK yes, I merged things even more.